### ISSUE TITLE
Warning: setState(...): Can only update a mounted or mounting component.

### ISSUE THREAD

**Issue by pierrecholhot** [ubilabs/react-geosuggest#71]
When the Geosuggest `input` is in focus and the react route changes, the following warning pops in the console:

> Warning: setState(...): Can only update a mounted or mounting component.
> This usually means you called setState() on an unmounted component.
> This is a no-op. Please check the code for the Geosuggest component.

Then it points to this part:

```js
/**
 * When the input loses focused
 */
hideSuggests: function hideSuggests() {
  this.props.onBlur();
  setTimeout((function () {
    this.setState({ isSuggestsHidden: true });
  }).bind(this), 100);
},
```

Not sure if I am supposed to `destroy` the component somehow in my `componentWillUnmount()` ?

----------------

![geo](https://cloud.githubusercontent.com/assets/1053458/11957044/2a2c8390-a8bf-11e5-927e-dfe867235b49.png)

----------------

```JSON
    "history": "~1.13.x",
    "react": "^0.14.3",
    "react-dom": "^0.14.3",
    "react-geosuggest": "^1.13.0",
    "react-router": "^1.0.2",
```

**Comment by ro-ka** [comment]
I suppose we need a check in that `setTimeout` whether it is still mounted or not.

**Comment by romseguy** [comment]
I'm still getting that warning, it looks like we can't call `this.setState` in `componentWillUnmount` here https://github.com/ubilabs/react-geosuggest/blob/master/src/Geosuggest.jsx#L90 

Related issue https://github.com/facebook/react/issues/2787 ; this comment particularly https://github.com/facebook/react/issues/2787#issuecomment-68738793

**Comment by ro-ka** [comment]
Will take a look.

**Comment by amitava82** [comment]
I just received same error.

**Comment by ro-ka** [comment]
Released in `1.19.0`.

**Comment by sergeyt** [comment]
This warning can still output in callback of autocomplete service, [now at line 190](https://github.com/ubilabs/react-geosuggest/blob/master/src/Geosuggest.jsx#L190).
